Lib/asyncio/base_events.py

```diff
@@ -302,6 +304,30 @@ def get_task_factory(self):
         """Return a task factory, or None if the default one is in use."""
         return self._task_factory
 
+    def current_task(self):
+        return self._current_task
+
+    def all_tasks(self):
+        return set(self._tasks)
+
+    def _register_task(self, task):
+        self._tasks.add(task)
+
+    def _enter_task(self, task):
+        if self._current_task is not None:
+            raise RuntimeError("Entring into task {!r} "
+                               "when other task {!r} is executed. "
+                               "The loop invariants are broken"
```

Comment:
No need for "The loop invariants are broken". I think it's pretty obvious when an event loop raises a RuntimeError.

Reply:
Done